Update project_idea.md (#9)
arviz-devs/GSoD: 2022/project_idea.md

# Centralized documentation for computational Bayesian workflow

## About ArviZ

> ArviZ is an open-source library dedicated to building tools for Exploratory Analysis of Bayesian Models. 

Reviewer: ```suggestion
ArviZ is an open-source project dedicated to building tools for Exploratory Analysis of Bayesian Models. 
```

I would keep project here. ArviZ can mean both project or library (either in python or in julia) but here we are using it as project. The ArviZ project maintains multiple libraries: ArviZ python, ArviZ julia, PSIS.jl, xarray-einstats... and educational resources, all dedicated to Exploratory Analysis of Bayesian Models.

Author: Alright, I understand now.

Statisticians and data analysts across both academic and industry, across programming languages 
and across probabilistic programming frameworks trust ArviZ to assist them to perform statistical workflows. 
They study a wide range of problems, from the effectiveness of cancer treatments 
to SpaceX rocket supply chains and even user behaviour on the internet.

To aid with Bayesian modeling analysis, ArviZ provides both numerical and visual summaries,
as well as a collection of software tools that help with modeling and storing statistical results.

ArviZ currently has a Julia and a Python interface and we are working on compatibility with R too.
To date of writing, ArviZ integrates with at least 9 different Probabilistic Programming Languages (PPLs),
including all major PPLs like Stan, PyMC or Turing.

### Contact 

> For any inquiry about Season of Docs at ArviZ, contact us at gitter.

Reviewer: ```suggestion
For any inquiry about Season of Docs at ArviZ, contact us at [gitter](https://gitter.im/arviz-devs/season_of_docs).
```

The gitter link is necessary

## Project motivation

> Writing educational contents to teach how to use ArviZ  could pose a certain level of difficulty, producing educational contents to teach
> how to use PPLs and Bayesian modeling could equally pose a level of difficulty. This difficulty isn't because these frameworks and libraries are inherently complex or too technical but rather as a result of having an encompassing documentation that captures all
> the possible libraries that are used with the Bayesian workflow
> without doing a disservice to any group of users by not capturing their choice of library or language. 
> Here's the icebreaker, writing such educational material/guide is actually one and the same, 
> a wider task generally called Bayesian workflow.[Bayesian workflow](https://arxiv.org/abs/2011.01808].

Reviewer: ```suggestion
Writing educational contents to teach how to use ArviZ is difficult, producing educational contents to teach
how to use PPLs or Bayesian modeling is also difficult. 
This difficulty isn't because these frameworks and libraries are inherently complex or too technical 
but rather as a result of having to encompass the use of several libraries and multiple theoretical topics.
Here's the key insight, the creation of educational material/guides on ArviZ, PPLs or Bayesian modeling
are entangled within them, it is actually one and the same, 
a wider task generally called Bayesian workflow.[Bayesian workflow](https://arxiv.org/abs/2011.01808].
```

(see also comment below)

> Solving real problems with the Bayesian model requires computational tools, 
> so trying to explain Bayesian modeling without using any PPL is very limited and not efficient. 
> But at the same time, explaining how to use a PPL to someone who doesn't know 
> anything about Bayesian modeling is also a fool's errand. 
> Any modeling without visualization, diagnostics and model checking is 

Reviewer: ```suggestion
Solving real problems with Bayesian modeling requires computational tools, 
so trying to explain Bayesian modeling without using any PPL is very limited and not efficient. 
But at the same time, explaining how to use a PPL to someone who doesn't know 
anything about Bayesian modeling is also a fool's errand. 
Moreover, any modeling without visualization, diagnostics and model checking (tasks done with ArviZ) is 
```

I have changed "the Bayesian model" with "Bayesian modeling", there isn't a single Bayesian model. "Bayesian modeling" is a way of understanding probability that allows statisticians to build any model and adapt them to the data
in order to get a good explanation of the observations.

This paragraph emphasizes with examples the general idea exposed in the paragraph above. I would then also move the ideas you suggested about the disservice to our users to a new paragraph right below this one. Let me know if this structure is making the idea we want to transmit a bit more clear.

Author: That's fine.

Author: > Thanks for the updates and for keeping the original lines heart it makes it much easier to review!
> 
> I have done some formatting changes that I'll commit myself. They will be marked as resolved, but you can still open them and see the changes and explanation in some of the cases. You'll have to pull if you are working locally.
> 
> Let me know if you don't know how to do it or would do a pair programming session to look at this or the reviews that need further discussion, you can reach me at writethedocs slack for example. I really appreciate you reviewing this proposal and making sure it is clear to people without any background on Bayesian statistics (like I expect all reviewers for where we submit this to try and get some funding to be)

I'm glad I could help, thank you.

doomed from the start.

> **We need to teach all these skills at the same time in an organized and coherent manner**.

Reviewer: ```suggestion
**We need to teach all these skills at the same time in an organized and coherent manner**.

In the case of ArviZ in particular, we also aim to integrate with all available PPLs instead of only one.
Therefore, we need to write documentation that captures all
the possible libraries that are used with the Bayesian workflow.
Otherwise we would be doing a disservice to many of our users by not capturing their choice of library or language. 
The resource proposed here however is not limited to using ArviZ with multiple PPLs,
we also aim to include an R version for example (where ArviZ is not available) and
use alternatives to ArviZ.
```

> In fact, this is already what is happening. For several years now, 

Reviewer: ```suggestion
In fact, the multi-PPL and multi-language approach is already what is happening. For several years now, 
```

all books in Bayesian modeling uses and explain computational tools, 
they very often have a "paired" GitHub repo with the code used in the book, 
_and the most popular books are "translated" to other PPLs and programming languages._ 
To give one example, [Statistical Rethinking](https://xcelab.net/rm/statistical-rethinking/)
has had their code examples translated into at least 3 programming languages 
(R, Python and Julia) and multiple PPLs. The book homepage lists 10 such translations. 
A 2 minute search to see if the book had been translated into Spanish only 
brought up Spanish bookstores selling the original English version of the book.

## About the project
Given ArviZ's position within the PPL ecosystem, some of the members of our team 
double as members of PPL development teams. We would like to create an open, collaborative, online 
and multi-language (in all its meanings!) educational resource about the Bayesian workflow. This resource will cover the various compatible libraries that can be used to create Bayesian workflow with sample visualizations and codes.

The goal of the project is to set up all the infrastructure for such a resource to 

> avoid duplications like the one encountered above, and once this is achieved, guide the ArviZ educational lead 

Reviewer: I am not sure what are you referring to with the "one encoutered above" is it the statistical rethinking book?

Author: yes, it is. If it is ambiguous or misunderstood, then you should scrap it.

so they can start writing content.

> ### Project’s scope

Reviewer: ```suggestion
## Project’s scope
```

To match the original hierarchy which came from the gsod template. "about the project" and "project scope" should be at the same level, not one inside the other.

Given ArviZ's position within the PPL ecosystem, some of the members of our team 
double as members of PPL development teams. We would like to create an open, collaborative, online 
and multi-language (in all its meanings!) educational resource about the Bayesian workflow. This resource will cover the various compatible libraries that can be used to create Bayesian workflow with sample visualizations and codes.

The goal of the project is to set up all the infrastructure for such a resource to 

> avoid duplications like the one encountered above, and once this is achieved, guide the ArviZ educational lead 

Reviewer: ```suggestion
avoid duplication and to avoid serving only a subset of the users, and once this is achieved, guide the ArviZ educational lead 
```

> - The bulk of the explanation/prose content will be common. 
>   The _multiple_ pages for each programming language and PPL need to be combined 
>   with the single explanation source of truth and rendered.
> - The code examples must be written in executable files and store their outputs
>   for rendering automatically (and without extra code). 
>   This will not be a static resource and will need to be updated regularly.
> - The explanation content must be written in a format that supports internationalization. 
>   Ideally, code annotations or comments would do but they can be avoided or kept in English.
> 
> - Changes to the explanation content should be reflected on the website automatically
>   when merged. Reexecuting and updating code outputs can have manual steps.
> * Set up  a proof of concept in the [bayesian-workflow](https://github.com/arviz-devs/bayesian-workflow) repo
> * together with all CI/CD necessary for updating the content.
> * Regular presentations to the ArviZ team about the writing workflow.

Reviewer: ```suggestion
  - The bulk of the explanation/prose content will be common. 
    The _multiple_ pages for each programming language and PPL need to be combined 
    with the single explanation source of truth and rendered.
  - The code examples must be written in executable files and store their outputs
    for rendering automatically (and without extra code). 
    This will not be a static resource and will need to be updated regularly.
  - The explanation content must be written in a format that supports internationalization. 
    Ideally, code annotations or comments would do but they can be avoided or kept in English.
  - Changes to the explanation content should be reflected on the website automatically
  when merged. Reexecuting and updating code outputs can have manual steps.
* Set up  a proof of concept in the [bayesian-workflow](https://github.com/arviz-devs/bayesian-workflow) repo
  together with all CI/CD necessary for updating the content.
* Regular presentations to the ArviZ team about the writing workflow.
```

Adding spaces to keep the bullet points with dashes as 2nd level bullet points inside the first level 1 bullet point.

* Write documentation about maintaining the repository.
* Define the style and structure of the resource.

> * Start adding some example/template pages (together with the ArviZ educational lead). 
> * For example, a quick way to add content and motivate other people to 
> * collaborate would be to adopt some case studies from the documentation 
> * of PPLs.

Reviewer: ```suggestion
* Start adding some example/template pages (together with the ArviZ educational lead). 
  For example, a quick way to add content and motivate other people to 
  collaborate would be to adopt some case studies from the documentation 
  of PPLs.
```

Keep this as text of the bullet point defined in the first line, not each line a different bullet point.

>   Work that is out of scope for this project

Reviewer: ```suggestion
Work that is out of scope for this project
```

* Writing the actual resource.

### Measuring your project’s success
Given the scope of the enterprise and its very early stage, we believe the best metrics are:
* Publishing the proof of concept (yes/no kind of metric)
* Guiding ~3 ArviZ team members or external collaborators through the process of adding new pages and translations, 
  so they can continue working independently.

> - We tried setting up some different educational resources a few years ago, but 
>   the writing workflow was very complicated, most of us were unable to 
>   generate previews and they were not automated either. Even if it is an 
>   unconventional metric, we believe it is key for the project's success 
>   that we, as the ones who will write and onboard external collaborators, 
>   understand the writing process and feel comfortable with it.

Reviewer: ```suggestion
  - We tried setting up some different educational resources a few years ago, but 
    the writing workflow was very complicated, most of us were unable to 
    generate previews and they were not automated either. Even if it is an 
    unconventional metric, we believe it is key for the project's success 
    that we, as the ones who will write and onboard external collaborators, 
    understand the writing process and feel comfortable with it.
```

>  We do not think that views are a valuable nor valid metric because it will be a resource in 
>  early stages of construction and won't have been advertised by then.

Reviewer: ```suggestion
We do not think that views are a valuable nor valid metric because it will be a resource in 
early stages of construction and won't have been advertised by then.
```

## Project budget

> Budget item
> 
> * Technical writer to create, update and review documentation of the infrastructure.
> $10,000
> * Volunteer stipend for mentor $500.
>  $500
> * Domain registration and renovation.
> $50
> * Downstream donation to infrastructural dependencies like sphinx to generate the website from the source docs, libraries like myst-nb to parse the jupyter notebooks as  sources for the website, readthedocs for building and hosting the website+translations, transifex for managing collaborative translations. 
>  $450
>  
> * Donations
>  Any extra funds left from the domain budget item will be used here.

Reviewer: ```suggestion
* Technical writer to create, update and review documentation of the infrastructure. $10,000
* Volunteer stipend for mentor $500.
* Domain registration and renovation. $50
* Downstream donation to infrastructural dependencies like sphinx to generate the website from the source docs, libraries like myst-nb to parse the jupyter notebooks as  sources for the website, readthedocs for building and hosting the website+translations, transifex for managing collaborative translations.  $450
  * Any extra funds left from the domain budget item will be used here and added to the "downstream donation" budget item.
```
